ledger/controllers/wallet.js

```diff
@@ -767,6 +797,74 @@ module.exports.initialize = async (debug, runtime) => {
   await runtime.queue.create('wallet-report')
 }
 
+function claimWalletHandler (runtime) {
+  return async (request, reply) => {
+    const debug = braveHapi.debug(module, request)
+    const { params, payload } = request
+    const { signedTx } = payload
+    const { paymentId } = params
+    const wallets = runtime.database.get('wallets', debug)
+    const members = runtime.database.get('members', debug)
+
+    const wallet = await wallets.findOne({ paymentId })
+    if (!wallet) {
+      return reply(boom.notFound())
+    }
+
+    const txn = runtime.wallet.validateTxSignature(wallet, signedTx, {
+      destinationValidator: braveJoi.string().guid(),
+      minimum: 0.001
+    })
+    const { postedTx } = await runtime.wallet.submitTx(wallet, txn, signedTx, false)
+    const {
+      type,
+      isMember,
+      node = { user: { id: '' } }
+    } = postedTx.destination
+    const memberId = node.user.id
+
+    // check that where the transfer is going to is a card, that belongs to a member
+    if (type !== 'card' || !isMember || !memberId) {
+      return reply(boom.forbidden())
+    }
+    // if wallet has already been claimed, don't tie wallet to member id
+    if (wallet.memberId) {
+      // Check if the member matches the associated member
+      if (memberId !== wallet.memberId) {
+        return reply(boom.forbidden())
+      }
+    } else {
+      await members.update({ memberId }, {
+        $set: {
+          memberId
+        }
+      }, { upsert: true })
+      // Attempt to associate the paymentId with the member, enforcing max associations no more than 3
+      const member = await members.findOneAndUpdate({
+        memberId,
+        $expr: {
+          $lt: [{
+            $size: {
+              $ifNull: ['$paymentIds', []]
+            }
+          }, 3]
+        }
+      }, {
+        $push: {
+          paymentIds: paymentId
+        }
+      })
+      // If association worked, perform reverse association, memberId with wallet
+      if (!member) {
+        return reply(boom.conflict())
+      }
+      await wallets.update({ paymentId }, { $set: { memberId } })
```

Comment:
@maikelmclauflin can we also set a property like `userCardId` which is the destination card here?